Cause.Unified and Cause.prettyPrint do not capture Throwable.cause

Not sure if it's the right solution, but can we add `cause: Unified` to `Unified` and adjust `Cause.prettyPrint` and maybe `Unified.toThrowable` accordingly? 

Btw `Unified.toThrowable` does not expose `className`, not sure how critical this is in practice for exception like:
```scala
class FooException(cause: Throwable) extend Exception(cause)
```
It seems most of information captured but such exception is lost.

Yeah, sorry that I didn't include a sample first. It seems in the code below we loose the cause stack trace:
```scala
object Example extends ZIOAppDefault {
  val run = ZIO.succeed(m0)

  def m0 = {
    m1
  }

  def m1 = try {
    m2
  } catch {
    case t: Throwable => throw new FooException(t)
  }

  def m2 = {
    m3
  }

  def m3 = {
    throw new Exception("Cause Error!")
  }
}

class FooException(cause: Throwable) extends Exception(cause)

//class FooException(cause: Throwable) extends Exception {
//  initCause(cause)
//}

object ExampleJava {
  def main(args: Array[String]): Unit = Example.m0
}
```
`Example` outputs:
```
timestamp=2022-04-01T12:33:03.879173Z level=ERROR thread=#zio-fiber-0 message="" cause="Exception in thread "zio-fiber-2" example.FooException: java.lang.Exception: Cause Error!
	at example.Example$.m1(CauseSpec.scala:15)
	at example.Example$.m0(CauseSpec.scala:9)
	at example.Example$.$anonfun$run$1(CauseSpec.scala:6)
	at zio.internal.FiberContext.runUntil(FiberContext.scala:242)
	at zio.internal.FiberContext.run(FiberContext.scala:116)
	at zio.internal.ZScheduler$$anon$1.run(ZScheduler.scala:151)
	at zio.internal.FiberContext.runUntil(FiberContext.scala:539)"
```
add missing `at example.Example$.m3(CauseSpec.scala:29)` compared to `ExampleJava` ouput:
```
Exception in thread "main" example.FooException: java.lang.Exception: Cause Error!
	at example.Example$.m1(CauseSpec.scala:21)
	at example.Example$.m0(CauseSpec.scala:15)
	at example.ExampleJava$.main(CauseSpec.scala:41)
	at example.ExampleJava.main(CauseSpec.scala)
Caused by: java.lang.Exception: Cause Error!
	at example.Example$.m3(CauseSpec.scala:29)
	at example.Example$.m2(CauseSpec.scala:25)
	at example.Example$.m1(CauseSpec.scala:19)
	... 3 more
```
And I hit another case where the cause exception is attached via `initCause` (see the commented block, not sure why there is a difference) but the ZIO output is missing any details of the cause exception in this case:
```
timestamp=2022-04-01T12:38:53.166458Z level=ERROR thread=#zio-fiber-0 message="" cause="Exception in thread "zio-fiber-2" example.FooException: null
	at example.Example$.m1(CauseSpec.scala:15)
	at example.Example$.m0(CauseSpec.scala:9)
	at example.Example$.$anonfun$run$1(CauseSpec.scala:6)
	at zio.internal.FiberContext.runUntil(FiberContext.scala:242)
	at zio.internal.FiberContext.run(FiberContext.scala:116)
	at zio.internal.ZScheduler$$anon$1.run(ZScheduler.scala:151)
	at zio.internal.FiberContext.runUntil(FiberContext.scala:539)"
```
vs
```
Exception in thread "main" example.FooException
	at example.Example$.m1(CauseSpec.scala:15)
	at example.Example$.m0(CauseSpec.scala:9)
	at example.ExampleJava$.main(CauseSpec.scala:35)
	at example.ExampleJava.main(CauseSpec.scala)
Caused by: java.lang.Exception: Cause Error!
	at example.Example$.m3(CauseSpec.scala:23)
	at example.Example$.m2(CauseSpec.scala:19)
	at example.Example$.m1(CauseSpec.scala:13)
	... 3 more
```

--- a/core/shared/src/main/scala/zio/Cause.scala
+++ b/core/shared/src/main/scala/zio/Cause.scala
@@ -507,11 +507,21 @@ sealed abstract class Cause[+E] extends Product with Serializable { self =>
       def unifyFail(fail: Cause.Fail[E]): Unified =
         Unified(fail.trace.fiberId, fail.value.getClass.getName(), fail.value.toString(), fail.trace.toJava)
 
-      def unifyDie(die: Cause.Die): Unified = {
-        val extra =
-          if (stackless) Chunk.empty else Chunk.fromArray(die.value.getStackTrace())
+      def unifyDie(die: Cause.Die): List[Unified] = {
 
-        Unified(die.trace.fiberId, die.value.getClass.getName(), die.value.getMessage(), extra ++ die.trace.toJava)
+        @tailrec
+        def loop(die: Cause.Die, result: List[Unified]): List[Unified] = {
+          val extra =
+            if (stackless) Chunk.empty else Chunk.fromArray(die.value.getStackTrace())
+
+          val unified =
+            Unified(die.trace.fiberId, die.value.getClass.getName(), die.value.getMessage(), extra ++ die.trace.toJava)
+
+          if (die.value.getCause eq null) unified :: result
+          else loop(Cause.Die(die.value.getCause, ZTrace.none), unified :: result)
+        }
+
+        loop(die, Nil)
       }
 
       def unifyInterrupt(interrupt: Cause.Interrupt): Unified = {
@@ -528,7 +538,7 @@ sealed abstract class Cause[+E] extends Product with Serializable { self =>
         case Stackless(cause, stackless) :: more => loop(cause :: more, fiberId, stackless, result)
         case Then(left, right) :: more           => loop(left :: right :: more, fiberId, stackless, result)
         case (cause @ Fail(_, _)) :: more        => loop(more, fiberId, stackless, unifyFail(cause) :: result)
-        case (cause @ Die(_, _)) :: more         => loop(more, fiberId, stackless, unifyDie(cause) :: result)
+        case (cause @ Die(_, _)) :: more         => loop(more, fiberId, stackless, unifyDie(cause) ::: result)
         case (cause @ Interrupt(_, _)) :: more =>
           loop(more, fiberId, stackless, unifyInterrupt(cause) :: result)
       }

